pkg/armrpc/asyncoperation/worker/worker.go

```diff
@@ -266,8 +265,7 @@ func (w *AsyncRequestProcessWorker) runOperation(ctx context.Context, message *q
 		if !errors.Is(asyncReqCtx.Err(), context.Canceled) {
 			if err != nil {
 				armErr := extractError(err)
-				result.SetFailed(armErr, false)
-				logger.Error(err, "Operation Failed")
```

Comment:
This is required for server side debugging. If we remove this, it is hard to debug only with server log. e.g. when we debug the problem with server log, it is hard to know which step has failed because there is no indication. Operation Failed will be the indicator at which step has been failed.

Reply:
This is not being removed in this PR, but being abstracted out. I have explained it in the PR description.

Reply:
This is not being removed in this PR, but being abstracted out as I also had explained in the PR description.